src/aequitas/squarify_flipped.py

```diff
@@ -0,0 +1,198 @@
+    # sizes should be sorted biggest to smallest
+    sizes = list(map(float, sizes))
+
+    if len(sizes) == 0:
+        return []
+
+    if len(sizes) == 1:
+        return layout(sizes, x, y, dx, dy)
+
+    # figure out where 'split' should be
+    i = 1
+    while i < len(sizes) and worst_ratio(sizes[:i], x, y, dx, dy) >= worst_ratio(sizes[:(i + 1)], x, y, dx, dy):
+        i += 1
+    current = sizes[:i]
+
+    remaining = sizes[i:]
+
+    (leftover_x, leftover_y, leftover_dx, leftover_dy) = leftover(current, x, y, dx, dy)
+
+    return layout(current, x, y, dx, dy) + \
+       squarify(remaining, leftover_x, leftover_y, leftover_dx, leftover_dy)
+
+
+def pad_rectangle(rect):
+    if rect['dx'] > 2:
+        rect['x'] += 1
+        rect['dx'] -= 2
+    if rect['dy'] > 2:
+        rect['y'] += 1
+        rect['dy'] -= 2
+
+
+def padded_squarify(sizes, x, y, dx, dy):
+    rects = squarify(sizes, x, y, dx, dy)
+    for rect in rects:
+        pad_rectangle(rect)
+    return rects
+
+def squarify_plot_rects(rects, norm_x=100, norm_y=100, color=None,
+                        label=None, value=None, ax=None, **kwargs):
+    """
+    Plotting with Matplotlib from predefined rectangles. Adapted from squarify
+    source code.
+
+    :param rects: list-like of dictionaries indicating rectangle dimensions
+        for plotting
+    :param norm_x:  overall figure dimensions for normalizing value box sizes
+    :param norm_y: overall figure dimensions for normalizing value box sizes
+    :param color: color string or list-like of colors to use for value boxes
+    :param label: list-like used as label text
+    :param value: list-like used as value text
+    :param ax: Matplotlib Axes instance
+    :param kwargs: dict, keyword arguments passed to matplotlib.Axes.bar
+
+    :return: matplotlib.Axis
+    """
+    x = [rect['x'] for rect in rects]
+    y = [rect['y'] for rect in rects]
+    dx = [rect['dx'] for rect in rects]
+    dy = [rect['dy'] for rect in rects]
+
+    ax.bar(x, dy, width=dx, bottom=y, color=color,
+           label=label, align='edge', **kwargs)
+
+    INDENT_LENGTH = 4
+    CHAR_PLACEHOLDER = 1.5
+
+    if value is not None:
+        va = 'center' if label is None else 'top'
+        for v, r in zip(value, rects):
+            if isinstance(v, (int, float)):
+                v = f"{v:.2}"
+
+            x, y, dx, dy = r['x'], r['y'], r['dx'], r['dy']
+
+            # if box large enough, add labels and values
+            if (dx >= (INDENT_LENGTH * 2) + (CHAR_PLACEHOLDER * len(v))) & (dx > 10):
+                ax.text(x + dx / 2, y + dy / 2, v, va=va,
+                        ha='center', fontsize=14)
+
+    if label is not None:
+        va = 'center' if value is None else 'bottom'
+        under_plot = []
+        alphabet = list(map(chr, range(65, 91)))
+        under_plot_num = 0
+
+        for i, (l, r) in enumerate(zip(label, rects)):
+            x, y, dx, dy = r['x'], r['y'], r['dx'], r['dy']
+            length = dx
```

Comment:
unclear what the enumeration is doing for you here

Reply:
used it to index list 'plot_ready' for value labels in line 191, but let me know if there's a more elegant way of accessing that!

Reply:
ah! i see now. that does seem like an inefficient and opaque way of doing this.

as far as I can tell, your `value` here doesn't change across iterations of your loop -- so why re-map it to `plot_ready` over and over for each iteration? (_and_ then require an index with which to retrieve from that mapped list the one item you need?)

for this, you might just map `value` once, and then change your loop like so:

```
import itertools

…

if value:
    plot_ready_values = [
        val if isinstance(val, str)
        else f"{val:.2}" if isinstance(val, (int, float))
        else ""
        for val in value
    ]
else:
    plot_ready_values = itertools.repeat(None)

for (l, r, val) in zip(label, rects, plot_ready_values):
    …
    underplot_label = str(l) if val is None else f"{l}, {val}"
```

Reply:
another thing, which just made it difficult to figure out what was going on in this function -- single-character variable names are _pretty_ evil. there are certainly cases where it makes sense to use names like "x" and "y"; (and, these at least relatively stand out). but by far it's best to give all variables longer, more descriptive names, generally at least three characters long or so, (though of course that needn't always be the case).

for that matter, it seemed a little mean to name "list-like" arguments with anything but plural names -- "value" and "label". These _look_ very much like non-iterable objects, judging by their names; yet, they're treated as though they are invariably iterable.

Reply:
Noted! I plan to eventually do a pull request for the library so didn't want to make too too many changes, but these make sense, improve clarity, and reduce unnecessary intermediate data structures, so I'm all for them.

Reply:
Cool :smile: